frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/form-switcher-validation.ts

```diff
@@ -0,0 +1,97 @@
+import * as _ from 'lodash';
+import { ValidationError } from 'yup';
+import { safeJSToYAML } from '@console/shared/src/utils/yaml';
+import { PipelineKind } from '../../../types';
+import { initialPipelineFormData } from './const';
+import { pipelineBuilderYAMLSchema } from './switch-to-form-validation-utils';
+import { PipelineBuilderFormValues } from './types';
+import { convertBuilderFormToPipeline } from './utils';
+
+export const getFormData = (
+  formData: PipelineBuilderFormValues,
+  yamlPipeline: PipelineKind,
+): PipelineBuilderFormValues => {
+  const { finally: finallyTasks, ...specProps } = yamlPipeline.spec;
+  const pipelineSpecProperties = _.omit(specProps, ['listTasks', 'finallyListTasks']);
+  return {
+    ...formData,
+    ...pipelineSpecProperties, // support & keep unknown values as well as whatever they may have changed that we use
+    name: yamlPipeline.metadata?.name,
+    finallyTasks,
+  };
+};
+
+export const validateFormData = (
+  newFormData: PipelineBuilderFormValues,
+  yamlPipeline: PipelineKind,
+) =>
+  pipelineBuilderYAMLSchema(newFormData).validate(yamlPipeline, {
+    abortEarly: false,
+    strict: true,
+  });
+
+export const safeOmit = (object: PipelineKind, path: string): PipelineKind => {
+  // Check if it's a `path.to.array[#]`
+  const match = path.match(/(.*)\[(\d+)\]$/);
+  if (match) {
+    const [, pathToArray, index] = match;
+    if (pathToArray && index) {
+      return _.update(object, pathToArray, (array) => {
+        array.splice(index, 1);
+        return array;
+      });
+    }
+  }
+  return _.omit(object, path);
+};
+
+export const handleSanitizeToFormError = async (
+  formData: PipelineBuilderFormValues,
+  error: ValidationError,
+  yamlPipeline: PipelineKind,
+): Promise<PipelineBuilderFormValues> => {
+  const paths: string[] = error?.inner?.map((err) => err.path).filter((path) => !!path) || [];
+  const validatedYamlData: PipelineKind = paths.reduce((acc, currPath) => {
+    const pathComponents = currPath.split('.');
+    const lastPathItem: string = pathComponents[pathComponents.length - 1];
+    switch (lastPathItem) {
+      case 'name': {
+        const namePath = pathComponents.slice(0, pathComponents.length - 1).join('.');
+        return safeOmit(acc, namePath);
+      }
+      default:
+        return safeOmit(acc, currPath);
+    }
+  }, yamlPipeline);
+  const validatedFormData = getFormData(formData, validatedYamlData);
+  try {
+    await validateFormData(validatedFormData, validatedYamlData);
+  } catch (err) {
+    return undefined;
+  }
+  return _.merge({}, initialPipelineFormData, validatedFormData);
```

Comment:
I think this `validatedFormData` needs to be sanitized again null values. If you remove all the value in tasks field and switch to pipeline builder, it breaks the console.
![task_default_value](https://user-images.githubusercontent.com/9964343/125790311-c8032901-a136-4bf5-ab0c-e436854e9646.gif)

Reply:
so I tried this out for event sources as well. it seems to be a problem for all components `SyncFieldEditor`.
I feel this should be handled by the component using the `SyncFieldEditor`. I'm not sure the fact that this is a common problem for all components changes that. (as far as I tested it is reproducible in Pipelines and Event sources, but I'm pretty sure all other components will face this problem as well)

Reply:
So since the `SyncEditorField` doesn't have knowledge of the YAML structure, it has to be handled by the component using it, which means apparently all the components using the `SyncEditorField` now has a bug.